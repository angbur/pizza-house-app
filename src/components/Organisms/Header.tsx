import { PropsWithChildren } from 'react';
import Button from '../Atoms/Button/Button';
import { Theme } from '../Theme/theme.types';

type HeaderProps = {
  theme: Theme;
};

const Header = ({ theme }: PropsWithChildren<HeaderProps>) => {
  return (
    <Button variant={'button-text-light'} theme={theme}>
      Press me
    </Button>
  );
};

export default Header;
