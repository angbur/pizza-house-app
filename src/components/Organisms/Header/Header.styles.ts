import { getInitialTheme } from './../../Theme/initialTheme';
import styled from 'styled-components';

export const Navbar = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0;
  margin: 0;
  height: 138px;
  width: 100vw;
  border-bottom: 2px solid ${getInitialTheme().palette.light};
  background: ${getInitialTheme().palette.light};
  box-shadow: 0px 2px 4px -1px rgb(0 0 0 / 20%), 0px 4px 5px 0px rgb(0 0 0 / 14%),
    0px 1px 10px 0px rgb(0 0 0 / 12%);
  & .header_logo {
    flex: 40%;
    display: flex;
    justify-content: flex-start;
    @media (max-width: 768px) {
      flex: 20%;
    }
  }
  & .header_title {
    flex: 20%;
    display: flex;
    justify-content: center;
    @media (max-width: 768px) {
      display: none;
    }
  }
`;

export const HeaderActions = styled.div`
  flex: 40%;
  display: flex;,
  justify-content: center;
  align-items: center;
  gap: 2rem;
  margin-right: 2rem;
  justify-content: flex-end;
  @media (max-width: 768px) {
    flex: 80%;
    gap: 2rem;
  }
  & button.header_actions_button--desktop {
    @media (max-width: 460px) {
      display: none;
    }
  }
  & button.header_actions_button--mobile {
    display: none;
    @media (max-width: 460px) {
      display: block;
    }
  }
`;
