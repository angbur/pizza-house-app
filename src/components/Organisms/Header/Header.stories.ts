import { getInitialTheme } from './../../Theme/initialTheme';
import { Meta } from '@storybook/react';
import Header from './Header';

const meta = {
  title: 'Organisms/Header',
  component: Header,
  tags: ['docsPage'],
  argTypes: {
    theme: {
      defaultValue: getInitialTheme(),
    },
  },
} satisfies Meta<typeof Header>;

export default meta;
