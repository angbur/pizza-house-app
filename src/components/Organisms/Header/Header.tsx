import Button from 'components/Atoms/Button/Button';
import Logo from 'components/Atoms/Logo/Logo';
import { HeaderActions, Navbar } from './Header.styles';
import Title from 'assets/image/quote.svg';
import LoginIcon from 'assets/icon/user.svg';
import { ThemeContext } from 'components/Theme/ThemeContext';
import { useContext } from 'react';
import { Link } from 'react-router-dom';

const Header = () => {
  const theme = useContext(ThemeContext);

  return (
    <header>
      <Navbar className='header' theme={theme}>
        <Link to={'/'} className='header_logo'>
          <Logo background='secondary' />
        </Link>
        <div className='header_title'>
          <img src={Title} />
        </div>
        <HeaderActions className='header_actions'>
          <Button variant={'primary-light'} className={'header_actions_button--desktop'}>
            Sign in
          </Button>
          <Button variant={'button-text-dark'} className={'header_actions_button--desktop'}>
            Sign up
          </Button>
          <Button variant={'icon-button'} className={'header_actions_button--mobile'}>
            <img src={LoginIcon} />
          </Button>
        </HeaderActions>
      </Navbar>
    </header>
  );
};

export default Header;
