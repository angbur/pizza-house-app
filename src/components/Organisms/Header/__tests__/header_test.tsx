import Header from '../Header';
import { renderWithProviders } from 'test/test-utils';

describe('Header', () => {
  it('should renders header correctly for a non-logged in user', () => {
    const renderedHeader = renderWithProviders(<Header />, {}).container;

    expect(renderedHeader).toMatchSnapshot();
  });

  it('should renders header correctly for a logged in user', async () => {
    const initialState = {
      loggedIn: true,
      status: '200',
      error: undefined,
      userId: 'dummy_id',
      login: 'dummy_login',
      token: 'dummy_token',
    };

    const header = renderWithProviders(<Header />, {
      preloadedState: {
        user: initialState,
      },
    }).container;

    expect(header).toMatchSnapshot();
  });
});
