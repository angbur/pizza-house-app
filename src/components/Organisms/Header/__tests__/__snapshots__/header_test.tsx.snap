// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should renders header correctly for a logged in user 1`] = `
<div>
  <header>
    <div
      class="sc-kDvujY hragjb header"
    >
      <a
        class="header_logo"
        href="/"
      >
        <div
          class="sc-jrcTuL cFiwzV full logo"
          style="background: rgb(21, 21, 21);"
        >
          <img
            src="logo.svg"
          />
        </div>
      </a>
      <div
        class="header_title"
      >
        <img
          src="quote.svg"
        />
      </div>
      <div
        class="sc-ipEyDJ cXcjSP header_actions"
      >
        <div
          class="sc-bqWxrE eaVRud"
        >
          <button
            class="sc-bcXHqe sc-jSUZER feNVH dkmCUq button-text-without-hover btn-lg button"
            style="padding: 0.65rem 1.25rem; font-size: 1.05rem;"
          >
            <h5
              class="item-title"
              style="font-family: Source Sans Pro; font-weight: 600; font-size: 1rem; line-height: 1.4; letter-spacing: 1px; color: rgb(21, 21, 21);"
            >
              dummy_login
            </h5>
          </button>
          <div
            class="sc-ksBlkl hTPapp"
            style="display: none;"
          >
            <li>
              <button
                class="sc-bcXHqe sc-jSUZER feNVH dkmCUq button-text-without-hover btn-lg button"
                style="padding: 0.65rem 1.25rem; font-size: 1.05rem;"
              >
                <img
                  height="14"
                  src="[object Object]"
                />
                 Profile
              </button>
            </li>
            <li>
              <button
                class="sc-bcXHqe sc-jSUZER feNVH dkmCUq button-text-without-hover btn-lg button"
                style="padding: 0.65rem 1.25rem; font-size: 1.05rem;"
              >
                <img
                  height="14"
                  src="[object Object]"
                />
                 Log out
              </button>
            </li>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header should renders header correctly for a non-logged in user 1`] = `
<div>
  <header>
    <div
      class="sc-kDvujY hragjb header"
    >
      <a
        class="header_logo"
        href="/"
      >
        <div
          class="sc-jrcTuL cFiwzV full logo"
          style="background: rgb(21, 21, 21);"
        >
          <img
            src="logo.svg"
          />
        </div>
      </a>
      <div
        class="header_title"
      >
        <img
          src="quote.svg"
        />
      </div>
      <div
        class="sc-ipEyDJ cXcjSP header_actions"
      >
        <button
          class="sc-bcXHqe sc-gswNZR feNVH gHullA primary-light btn-lg header_actions_button--desktop button"
          style="padding: 0.65rem 1.25rem; font-size: 1.05rem;"
        >
          Sign in
        </button>
        <button
          class="sc-bcXHqe sc-gKPRtg sc-ftTHYK feNVH hUzBlu hIdUJX button-text-dark btn-lg header_actions_button--desktop button"
          style="padding: 0.65rem 1.25rem; font-size: 1.05rem;"
        >
          Sign up
        </button>
        <div
          class="sc-eDWCr hgSJeQ"
        >
          <button
            class="sc-bcXHqe sc-jSUZER feNVH dkmCUq button-text-without-hover btn-lg button"
            style="padding: 0.65rem 1.25rem; font-size: 1.05rem;"
          >
            <img
              height="20"
              src="[object Object]"
            />
          </button>
          <div
            class="sc-ksBlkl hTPapp"
            style="display: none;"
          >
            <li>
              <button
                class="sc-bcXHqe sc-jSUZER feNVH dkmCUq button-text-without-hover btn-lg button"
                style="padding: 0.65rem 1.25rem; font-size: 1.05rem;"
              >
                Sign in
              </button>
            </li>
            <li>
              <button
                class="sc-bcXHqe sc-jSUZER feNVH dkmCUq button-text-without-hover btn-lg button"
                style="padding: 0.65rem 1.25rem; font-size: 1.05rem;"
              >
                Sign up
              </button>
            </li>
          </div>
        </div>
      </div>
    </div>
  </header>
</div>
`;
